tools/syz-declextract/syz-declextract.cpp

```diff
@@ -64,6 +106,230 @@ bool beginsWith(const std::string_view &str, const std::string_view begin) {
+    case clang::Type::Typedef:
+      return getFieldType(llvm::dyn_cast<TypedefType>(field)->desugar(), context, fieldType.getAsString(), parent,
+                          isSyscallParam);
+    case clang::Type::IncompleteArray: // Defined as type[]
+      return makeArray(getFieldType(llvm::dyn_cast<IncompleteArrayType>(field)->getElementType(), context, fieldName));
+    case clang::Type::ConstantArray: {
+      const auto &array = llvm::dyn_cast<ConstantArrayType>(field);
+      return makeArray(getFieldType(array->getElementType(), context, fieldName), array->getSize().getZExtValue());
+    }
+    case clang::Type::Pointer: {
+      const auto &pointerType = llvm::dyn_cast<PointerType>(field);
+      const auto &pointeeType = pointerType->getPointeeType();
+      const auto &fieldType =
+          pointeeType->isAnyCharacterType() ? "string" : getFieldType(pointeeType, context, fieldName);
+      const auto &ptrDir = pointeeType.isConstQualified() ? "in" : "inout"; // TODO: Infer direction of non-const.
+      return makePtr(ptrDir, fieldType,
+                     parent + "$auto_record" == fieldType); // Checks if the direct parent is the same as the node.
+    }
+    case clang::Type::Builtin: {
+      const auto &type = getSyzType(field.getAsString(), isSyscallParam);
+      return isPadding(fieldName) ? makeConst(isSyscallParam ? "" : type) : type;
+    }
+    case clang::Type::CountAttributed: // Has the attribute counted_by. Handled by getCountedBy
+      return getFieldType(llvm::dyn_cast<CountAttributedType>(field)->desugar(), context, fieldName);
+    case clang::Type::BTFTagAttributed: // Currently Unused
+      return getFieldType(llvm::dyn_cast<BTFTagAttributedType>(field)->desugar(), context, fieldName);
+    case clang::Type::Elaborated:
+      return getFieldType(llvm::dyn_cast<ElaboratedType>(field)->desugar(), context, fieldName, parent, isSyscallParam);
+    case clang::Type::Enum: {
+      const auto &enumDecl = llvm::dyn_cast<EnumType>(field)->getDecl();
+      auto name = enumDecl->getNameAsString();
+      flags.push_back(name);
+      includes.push_back(std::filesystem::relative(SM->getFilename(enumDecl->getSourceRange().getBegin()).str()));
+      const char *sep = " = ";
+      for (const auto &enumerator : enumDecl->enumerators()) {
+        flags.back() += sep + enumerator->getNameAsString();
+        sep = ", ";
+      }
+      return "flags[" + name + "]";
+    }
+    case clang::Type::FunctionProto:
+      return makePtr("in", "int8");
+    default:
+      field->dump();
+      fprintf(stderr, "Unhandled field type %s\n", field->getTypeClassName());
+      exit(1);
+    }
+  }
+
+  std::string extractRecord(const RecordDecl *recordDecl, ASTContext *context, const std::string &backupName = "") {
+    recordDecl = recordDecl->getDefinition();
+    if (!recordDecl) { // When the definition is in a different translation unit.
+      return "intptr";
```

Comment:
This is just a stub and is effectively wrong, right. Same for function pointers that are transformed to `ptr[in, int8]`. Are there any other such cases? The problem is that you have this knowledge in your head, but tomorrow it's lost.
It may be useful to add something like:
```
type auto_todo intptr
```
and then use `auto_todo` in all such cases. Then it's clear (here and in the generated descriptions) that it's not correct intptr, but rather a stub for something we don't handle properly yet.